### PR TITLE
[tests] time out tests in test_all after a duration

### PR DESCRIPTION
**Pull request by sunshowers**
We're seeing some tests in `test_all` hang forever (#4779). Set a reasonable
upper bound on test duration.

This will also cause stdout and stderr for failing tests to be printed. Doing
so on SIGTERM in general is tracked at
https://github.com/nextest-rs/nextest/issues/1208.

Also, bump up the required nextest version to 0.9.64 to make use of the
`binary_id` predicate.

### CODE REVIEW

File: .config/nextest.toml [hunk 1]
```diff
@@ -30,5 +30,11 @@ command = 'cargo run -p crdb-seed --profile test'
...
 [[profile.default.overrides]]
-filter = 'package(oximeter-db) and test(replicated)'
+filter = 'binary_id(oximeter-db::replicated)'
 test-group = 'clickhouse-cluster'
+
+[[profile.default.overrides]]
+filter = 'binary_id(omicron-nexus::test_all)'
+# As of 2023-01-08, the slowest test in test_all takes 196s on a Ryzen 7950X.
+# 900s is a good upper limit that adds a comfortable buffer.
+slow-timeout = { period = '60s', terminate-after = 15 }
```
Comment (smklein) [inline .config/nextest.toml:40]: It's not great, but I've seen tests on my laptop that take up to 1000 seconds -- the saga 'test_action_failure_can_unwind' tests are particularly slow, and definitely need optimizing.

I'm not even sure I'd say this timeout is "wrong" - that test needs fixing - but figured I'd mention it regardless

Comment (smklein) [inline .config/nextest.toml:40]: It's not great, but I've seen tests on my laptop that take up to 1000 seconds -- the saga 'test_action_failure_can_unwind' tests are particularly slow, and definitely need optimizing.

I'm not even sure I'd say this timeout is "wrong" - that test needs fixing - but figured I'd mention it regardless

**Review by smklein — APPROVED** [review]

**Comment by davepacheco** [comment]
FWIW there is a timeout already on the whole test run and we went with a few hours:
https://github.com/oxidecomputer/omicron/blob/cde9b1547ea928c1b6ce0c11bfd09690c2502549/.github/buildomat/build-and-test.sh#L62

Clulow's Lament applies here: it's a tradeoff between false positives (spurious test failures because we didn't wait long enough) vs. taking too long to get feedback when they're hung.  Personally, if a test hangs when I'm running it locally, I'd always rather it hang (and not kill it) because that will preserve a lot of useful runtime state I might want to debug it.  In CI I don't think I care either way.  But false positives here would certainly suck.

File: .config/nextest.toml (hunk 1 above)
Comment (sunshowers) [reply .config/nextest.toml:40]: Hmm, that test isn't part of `test_all`.

**Comment by sunshowers** [comment]
> In CI I don't think I care either way. But false positives here would certainly suck.

Fair, I'll restrict it to CI.

**Pull request merged**